File: Algoritmos de ordenamiento/cpp/02 Shuffle.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
using namespace std;

int main() {
    system("title Algoritmos de ordenamientos de datos en C++ / Shufflesort&& cls");

    int cache, cache2 = 0;

    int array[10] = {2, 3, 1, 923, -23, 9128, 0, 10, 20, 12};

    size_t size = sizeof(array)/sizeof(array[0]);

    srand(time(NULL));

    cout << "\n== Estos son tus valores == \n\n";

    for ( int i = 0; i < size; i++ ) {
        cout << array[i] << "	";
    }

    cout << "\n\n== Desordenando... ==\n\n";

    for ( int i = 0; i < size ; i++) {
        cache2 = rand() % (size-i) + i;

        int cache = array[i];

        array[i] = array[cache2];

        array[cache2] = cache;

        for ( int i = 0; i < size; i++ ) {
            cout << array[i] << "	";
        }
        cout << endl;

    }
    cout << "\n== Bubblesort ==\n\n";

    for ( int i = 0; i < size; i++ ) {

        for ( int j = 0; j < size - i; j++ ) {

            if ( array[j+1] < array[j] ) {

                int cache = array[j+1];

                array[j+1] = array[j];

                array[j] = cache;
                
            }

            for ( int i = 0; i < size; i++ ) {
                cout << array[i] << "	";
            }
            cout << endl;
        }
    }

    cout << "\n== Estos son tus valores ordenados ==\n\n";

    for ( int i = 0; i < size; i++ ) {
        cout << array[i] << "	";
    }
    
    cout << endl;
    cout << endl;

    system("pause");
    system("cls");
    
}

/*
-lstdc++
*/

File: Algoritmos de ordenamiento/py/02 Shuffle.py
import random

def shuffle_sort(array):
    size = len(array)
    for i in range(size):
        cache2 = random.randint(i, size-1)
        array[i], array[cache2] = array[cache2], array[i]

def bubble_sort(array):
    size = len(array)
    for i in range(size):
        for j in range(size - i - 1):
            if array[j+1] < array[j]:
                array[j], array[j+1] = array[j+1], array[j]

array = [2, 3, 1, 923, -23, 9128, 0, 10, 20, 12]

print("\n== Estos son tus valores == \n")
print(*array, sep="\t")

print("\n\n== Desordenando... ==\n")

shuffle_sort(array)

for i in range(len(array)):
    print(array[i], end="\t")
print()

print("\n== Bubblesort ==\n")

bubble_sort(array)

for i in range(len(array)):
    print(array[i], end="\t")
print()

print("\n== Estos son tus valores ordenados ==\n")
print(*array, sep="\t")

print("\n")

input("Presiona Enter para continuar...")
